adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java

```diff
@@ -184,15 +217,23 @@ void updateTokenCache(final String resource, final String clientId, final Authen
      */
     void removeTokenCacheItem(final TokenCacheItem tokenCacheItem, final String resource)
             throws AuthenticationException {
+        final CacheEvent cacheEvent = new CacheEvent(EventStrings.TOKEN_CACHE_DELETE);
+        cacheEvent.setRequestId(mTelemetryRequestId);
+        Telemetry.getInstance().startEvent(mTelemetryRequestId, EventStrings.TOKEN_CACHE_DELETE);
+
         final List<String> keys;
         final TokenEntryType tokenEntryType = tokenCacheItem.getTokenEntryType();
         switch (tokenEntryType) {
-        case REGULAR_TOKEN_ENTRY : 
+        case REGULAR_TOKEN_ENTRY :
+            cacheEvent.setTokenTypeRT(true);
             Logger.v(TAG, "Regular RT was used to get access token, remove entries "
                     + "for regular RT entries.");
             keys = getKeyListToRemoveForRT(tokenCacheItem);
             break;
-        case MRRT_TOKEN_ENTRY : 
+        case MRRT_TOKEN_ENTRY :
+            // We delete both MRRT and RT in this case.
+            cacheEvent.setTokenTypeMRRT(true);
+            cacheEvent.setTokenTypeRT(true);
```

Comment:
We not sure that RT was there right? So why do we log it here? I think it would be confusing to indicate that RT token was in the cache.